### ISSUE TITLE
feat(event-store): add guarantee to .once projections

### ISSUE THREAD

**Issue by kodemon** [valkyrjs/valkyr#2]
### What is needed?

Add a guarantee to .once projections.

### Why is this needed?

Currently a once projection is designed only to run the first time a new event is created at the source level. And since this only happens at that specific time we want to ensure that the projection is successfully executed at least once. This is important for things such as notifications, third party updates, and event sagas.

### How can this be done?

For this specific projection scenario we can look at it as a job that has to complete before we are satisfied. In that regard we can probably store the projection state of the event in a separate table, and in the event of failures we have a tracked source of events that is missing a successful run.

**Comment by kodemon** [comment]
This is no longer needed as guarantees are handled outside of the event store layer.
